src/macros.h

```diff
@@ -230,11 +230,11 @@
  * PTR2CHAR(): get character from pointer.
  */
 /* Get the length of the character p points to, including composing chars */
-#define MB_PTR2LEN(p)	    (has_mbyte ? (*mb_ptr2len)(p) : 1)
+#define MB_PTR2LEN(p)	    (has_mbyte ? (*mb_ptr2len)(p) : (*p == NUL ? 0 : 1))
```

Comment:
Hmm, now I think it should simply be:
```
#define MB_PTR2LEN(p) *mb_ptr2len(p)
```
... but then I don't really see the need for a macro anymore.
Same remark for other macros.